code/obj/machinery/status_display.dm

```diff
@@ -368,71 +372,125 @@
+	var/emotion = null //an icon state
+	var/message = null //displays on examine
+	var/face_color = null
+
+	var/datum/light/screen_glow
 
-// 		use_power(200)
+	New()
+		..()
+		face_image = image('icons/obj/status_display.dmi', icon_state = "")
+		glow_image = image('icons/obj/status_display.dmi', icon_state = "ai_glow")
+		back_image = image('icons/obj/status_display.dmi', icon_state = "ai_white")
 
-// 		update()
 
-// 	proc/update()
-// 		if (mode == 0) //Blank
-// 			UpdateOverlays(null, "emotion_img")
-// 			picture_state = null
-// 			return
 
-// 		if (mode == 1)	// AI emoticon
-// 			if (src.emotion)
-// 				src.set_picture(src.emotion)
-// 			return
+		if (map_settings.walls ==/turf/simulated/wall/auto/jen)
+			pixel_y = 32
+		else
+			pixel_y = 29
 
-// 		if (mode == 2)	// BSOD
-// 			set_picture("ai_bsod")
-// 			return
+		screen_glow = new /datum/light/point
+		screen_glow.set_brightness(0.45)
+		screen_glow.set_height(0.75)
+		screen_glow.attach(src)
 
-// 	proc/set_picture(var/state)
-// 		if (!state || state == picture_state)
-// 			return //Hoooly balls why was this not here before argh
-// 		picture_state = state
-// 		pic_image.icon_state = picture_state
-// 		UpdateOverlays(pic_image, "emotion_img")
+	disposing()
+		if (screen_glow)
+			screen_glow.dispose()
+		..()
 
-// 	get_desc()
-// 		..()
-// 		if (status & NOPOWER)
-// 			return
-// 		if (src.message)
-// 			. += "<br>It says: \"[src.message]\""
+	process()
+		if (status & NOPOWER)
+			UpdateOverlays(null, "emotion_img")
+			UpdateOverlays(null, "back_img")
+			UpdateOverlays(null, "glow_img")
+			return
+		use_power(200)
+		update()
+
+	proc/update()
+
+
+		if (mode == 0 || !owner) //Blank
+			UpdateOverlays(null, "emotion_img")
+			UpdateOverlays(null, "back_img")
+			UpdateOverlays(null, "glow_img")
+			screen_glow.disable()
+			return
+		else //All the non-face stuff goes in here
+			if (!screen_glow.enabled)
+				screen_glow.enable()
+
+			if (face_color != owner.faceColor)
+				face_color = owner.faceColor
+				back_image.color = face_color
+				UpdateOverlays(back_image, "back_img")
+				UpdateOverlays(glow_image, "glow_img", 1) //forced so it displays on top
+				UpdateOverlays(face_image, "emotion_img", 1) //idem
+
+				var/colors = GetColors(face_color)
+				screen_glow.set_color(colors[1] / 255, colors[2] / 255, colors[3] / 255)
+
+			message = owner.status_message
+			name = initial(name) + " ([owner.name])"
+
+		if (mode == 1)	// AI emoticon
+			if (src.emotion != owner.faceEmotion)
+				src.set_picture(owner.faceEmotion)
+				emotion = owner.faceEmotion
+			return
+
+	proc/set_picture(var/state)
+		if (!state)
+			return //Hoooly balls why was this not here before argh
+		face_image.icon_state = state
+		UpdateOverlays(face_image, "emotion_img")
+
+	get_desc()
+		..()
+		if (status & NOPOWER)
+			return
+		if (src.message)
+			. += "<br>[owner.name] says: \"[src.message]\""
+
+	verb/claim() //Captain said it's my turn on the status display
```

Comment:
I think AI could just claim the display by clicking on it instead? (Done thourgh overriding `attack_ai()`.)

Reply:
The more you know